kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -200,9 +206,31 @@ protected Tuple2<Protocol, Metadata> loadTableProtocolAndMetadata(
         // Load this lazily (as needed). We may be able to just use the hint.
         ColumnarBatch columnarBatch = null;
 
+        // Note that (1) we don't yet know if ICT is enabled on this table and (2) we only care
+        // about getting the ICT from the latest commit whose version matches our desired snapshot
+        // version.
+        if (version == snapshotVersion) {
+          columnarBatch = nextElem.getColumnarBatch();
+          assert (columnarBatch.getSchema().equals(SNAPSHOT_DESCRIPTOR_LOG_REPLAY_SCHEMA));
+          final ColumnVector commitInfoVector = columnarBatch.getColumnVector(2);
+
+          for (int i = 0; i < commitInfoVector.getSize(); i++) {
+            if (!commitInfoVector.isNullAt(i)) {
+              ictOpt = CommitInfo.inCommitTimestampFromColumnVector(commitInfoVector, i);
+            }
+
+            if (!nextElem.isFromCheckpoint()) {
```

Comment:
1. commit info will never be read from checkpoint
2. make a call to read that .json file?

Reply:
cc @dhruvarya-db what do you think?

Reply:
Sounds right. Just confirming, I don't see that being done in this PR --- you are confirming before adding that right?

Reply:
> you are confirming before adding that right?

Yeah.